lib/segment/tests/integration/multivector_filtrable_hnsw_test.rs

```diff
@@ -0,0 +1,301 @@
+            },
+        )]),
+        sparse_vector_data: Default::default(),
+        payload_storage_type: Default::default(),
+    };
+
+    let int_key = "int";
+
+    let mut segment = build_segment(dir.path(), &config, true).unwrap();
+    for n in 0..num_points {
+        let idx = n.into();
+        let multi_vec = random_multi_vector(&mut rnd, vector_dim, num_vector_per_points);
+
+        let int_payload = random_int_payload(&mut rnd, num_payload_values..=num_payload_values);
+        let payload: Payload = json!({int_key:int_payload,}).into();
+
+        let named_vectors = only_default_multi_vector(&multi_vec);
+        segment
+            .upsert_point(n as SeqNumberType, idx, named_vectors)
+            .unwrap();
+        segment
+            .set_full_payload(n as SeqNumberType, idx, &payload)
+            .unwrap();
+    }
+
+    let payload_index_ptr = segment.payload_index.clone();
+
+    let hnsw_config = HnswConfig {
+        m,
+        ef_construct,
+        full_scan_threshold,
+        max_indexing_threads: 2,
+        on_disk: Some(false),
+        payload_m: None,
+    };
+
+    let permit_cpu_count = num_rayon_threads(hnsw_config.max_indexing_threads);
+    let permit = Arc::new(CpuPermit::dummy(permit_cpu_count as u32));
+
+    let vector_storage = &segment.vector_data[DEFAULT_VECTOR_NAME].vector_storage;
+    let quantized_vectors = &segment.vector_data[DEFAULT_VECTOR_NAME].quantized_vectors;
+    let mut hnsw_index = HNSWIndex::<GraphLinksRam>::open(
+        hnsw_dir.path(),
+        segment.id_tracker.clone(),
+        vector_storage.clone(),
+        quantized_vectors.clone(),
+        payload_index_ptr.clone(),
+        hnsw_config,
+    )
+    .unwrap();
+
+    hnsw_index.build_index(permit.clone(), &stopped).unwrap();
+
+    payload_index_ptr
+        .borrow_mut()
+        .set_indexed(&path(int_key), PayloadSchemaType::Integer.into())
+        .unwrap();
+    let borrowed_payload_index = payload_index_ptr.borrow();
+    let blocks = borrowed_payload_index
+        .payload_blocks(&path(int_key), indexing_threshold)
+        .collect_vec();
+    for block in blocks.iter() {
+        assert!(
+            block.condition.range.is_some(),
+            "only range conditions should be generated for this type of payload"
+        );
+    }
+
+    let mut coverage: HashMap<PointOffsetType, usize> = Default::default();
+    let px = payload_index_ptr.borrow();
+    for block in &blocks {
+        let filter = Filter::new_must(Condition::Field(block.condition.clone()));
+        let points = px.query_points(&filter);
+        for point in points {
+            coverage.insert(point, coverage.get(&point).unwrap_or(&0) + 1);
+        }
+    }
+    let expected_blocks = num_points as usize / indexing_threshold * 2;
+
+    assert!(
+        (blocks.len() as i64 - expected_blocks as i64).abs() <= 3,
+        "real number of payload blocks is too far from expected"
+    );
+
+    assert_eq!(
```

Comment:
We can remove coverage check, I guess it was copied from another test. But you can keep it if you wish